slick-testkit/src/main/scala/com/typesafe/slick/testkit/util/StandardTestDBs.scala

```diff
@@ -292,6 +295,9 @@ object DerbyDB {
 abstract class HsqlDB(confName: String) extends InternalJdbcTestDB(confName) {
   val profile = HsqldbProfile
   val jdbcDriver = "org.hsqldb.jdbcDriver"
+  // Hsqldb has valid "select for update" syntax, but in testing, it either takes a whole table lock or no exclusive
+  // lock at all, so exclude from ForUpdate testing
+  override def capabilities = super.capabilities - TestDB.capabilities.selectForUpdateRowLocking
```

Comment:
I couldn't work out why the test wasn't passing for Hsqldb. It's not a backend I'm familiar with. It could just be a configuration problem in the test environment. The syntax generated is valid and as far as I can tell, it should work. I took the decision to add the capability to the profile, but exclude from testing in case it was just a test configuration problem. I can exclude from the Hsql profile if you think that is better.